cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -150,6 +151,16 @@ public class AppMetadataStore {
       .put(ProgramRunStatus.REJECTED, TYPE_RUN_RECORD_COMPLETED)
       .build();
 
+  private static final String TYPE_FLOW_CONTROL_LAUNCHING = "flowControlLaunching";
```

Comment:
Consider not using prefix in values.  This would save database space and generally it's clear that it's flow control from field name. Also do you want to differentiate NONE from NULL? Most run records would be in NONE state, so using null would save quite some space. Also assume we don't do migrations, so counters would be off for any programs running while upgrading and that's fine (there should not be any). But having NULL would save you from updateing all old runs, all those would be consistent

Reply:
Removed. Done.